From b4b28617468c050174472c84845a0527e5626107 Mon Sep 17 00:00:00 2001
From: Li Huang <putin.li@rock-chips.com>
Date: Thu, 17 Aug 2023 08:23:11 +0000
Subject: [PATCH 03/14] SurfaceFlinger : Standard resolution switching method.

Fix a BUG for switching resolutin by standard method.
The destinationClip parameter will be abnormal.

Bug: Defect #303103
Change-Id: I283e848cc6963eaf7c7397f6062785d97ca895a1
Signed-off-by: Li Huang <putin.li@rock-chips.com>
Signed-off-by: libin <bin.li@rock-chips.com>
---
 services/surfaceflinger/SurfaceFlinger.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index b0575770ad..ea29a3758b 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -3938,6 +3938,12 @@ void SurfaceFlinger::processDisplayChanged(const wp<IBinder>& displayToken,
         if (currentState.flags != drawingState.flags) {
             display->setFlags(currentState.flags);
         }
+
+        if (currentState.width != drawingState.width ||
+            currentState.height != drawingState.height) {
+            display->setDisplaySize(currentState.width, currentState.height);
+        }
+
         if ((currentState.orientation != drawingState.orientation) ||
             (currentState.layerStackSpaceRect != drawingState.layerStackSpaceRect) ||
             (currentState.orientedDisplaySpaceRect != drawingState.orientedDisplaySpaceRect)) {
@@ -3951,7 +3957,6 @@ void SurfaceFlinger::processDisplayChanged(const wp<IBinder>& displayToken,
         }
         if (currentState.width != drawingState.width ||
             currentState.height != drawingState.height) {
-            display->setDisplaySize(currentState.width, currentState.height);
 
             if (display->getId() == mActiveDisplayId) {
                 onActiveDisplaySizeChanged(*display);
-- 
2.43.0

